cli/internal/cmd/upgradeapply.go

```diff
@@ -189,18 +190,25 @@ func (u *upgradeApplyCmd) upgradeApply(cmd *cobra.Command, upgradeDir string, fl
 		return fmt.Errorf("upgrading measurements: %w", err)
 	}
 
-	var tfOutput terraform.ApplyOutput
+	var infraState state.Infrastructure
 	if flags.skipPhases.contains(skipInfrastructurePhase) {
-		tfOutput, err = u.clusterShower.ShowCluster(cmd.Context(), conf.GetProvider())
+		infraState, err = u.clusterShower.ShowInfrastructure(cmd.Context(), conf.GetProvider())
 		if err != nil {
-			return fmt.Errorf("getting Terraform output: %w", err)
+			return fmt.Errorf("getting infra state: %w", err)
 		}
 	} else {
-		tfOutput, err = u.migrateTerraform(cmd, conf, upgradeDir, flags)
+		infraState, err = u.migrateTerraform(cmd, conf, upgradeDir, flags)
 		if err != nil {
 			return fmt.Errorf("performing Terraform migrations: %w", err)
 		}
 	}
+	state := state.State{
+		Version:        "v1",
+		Infrastructure: infraState,
+	}
+	if err := u.fileHandler.WriteYAML(constants.StateFilename, state, file.OptOverwrite); err != nil {
+		return fmt.Errorf("writing state file: %w", err)
+	}
```

Comment:
Consider moving the updating of an existing state file to `updateClusterIDFile` (rename the function eventually),
and have an explicit migration function if the state file does not exist when running `upgrade apply`.
IMO does not need to happen in this PR, but should be done before this is in any release.
If you decide to implement in a different PR, please add a comment to this section.